amethyst_network/src/bundle.rs

```diff
@@ -1,33 +1,34 @@
-use std::net::SocketAddr;
+use std::{marker::PhantomData, net::SocketAddr};
 
 use serde::{de::DeserializeOwned, Serialize};
 
 use amethyst_core::{bundle::SystemBundle, shred::DispatcherBuilder};
 use amethyst_error::{Error, ResultExt};
 
-use crate::{filter::NetFilter, server::ServerConfig, NetSocketSystem};
+use crate::{server::ServerConfig, NetSocketSystem};
 
 /// A convenience bundle to create the infrastructure needed to send and receive network messages.
 pub struct NetworkBundle<T> {
     /// the configuration used for the networking crate.
     config: ServerConfig,
-
-    /// The filters applied on received network events.
-    filters: Vec<Box<dyn NetFilter<T>>>,
+    _data: PhantomData<T>,
```

Comment:
this can be dropped and the `T` can be a bound on `fn build<T>`

Reply:
I am not sure what you mean I will have a 'unconstrained type parameter' error.

Reply:
You can drop the type T from this struct

Reply:
I can leave it away, but I can't add `T` do `fn build<T>` because it isn't part of the `SystemBundle` trait. 

"method `build` has 1 type parameter but its trait declaration has 0 type parameters"

Reply:
ahhh I forgot that its based on a trait you're right 👍